frodo.js

```diff
@@ -1,75 +1,81 @@
-/**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.